src/cache.c

```diff
@@ -1596,7 +1596,7 @@ cache_init(void)
 
   cmdbase = commands_base_new(evbase_cache, NULL);
 
-  ret = listener_add(cache_daap_listener_cb, LISTENER_DATABASE);
+  ret = listener_add(cache_daap_listener_cb, LISTENER_DATABASE | LISTENER_STICKER);
```

Comment:
I'll admit I don't really know what a sticker event is, but it seems odd that the cache regeneration needs for listen for it?

Reply:
Stickers are a concept in mpd to allow clients to store arbitrary key/vaue pairs to songs (https://www.musicpd.org/doc/protocol/stickers.html). If such a key/value pair changes, mpd issues a STICKER event. mpd does not support rating songs. mpd clients like Cantata or MPDroid therefor use stickers to support ratings (key = 'rating' and value between 1 to 10).

Updating the rating of a song did previously trigger the cache rebuild. As the rating update now triggers a STICKER event instead of a DATABASE event, the cache needs to listen to it (otherwise it would be a functional change).

Maybe it would be good to rename the event into 'LISTENER_RATING'?

Reply:
Don't know... I'm also thinking if cache rebuild is actually a good idea for updated ratings? Even if it was there before it may not have been very necessary. I can think of only one case where it might be, and it is a bit of an edge case: Requests from iTunes normally get cached, and since iTunes requests all tags at once it also gets the rating. So not updating the cache could return an incorrect rating.

It's up to you, but if you stick with listening for LISTENER_STICKER or _RATING then please add a comment that says a few words about why the cache is listening for it, since it is not entirely obvious.